linter.lua

```diff
@@ -25,10 +26,53 @@ local function run_lint_cmd(path, linter)
 end
 
 
+local function match_pattern(text, pattern, order)
+  if order == nil then
+    return text:gmatch(pattern)
+  end
+
+  return coroutine.wrap(function()
+    for one, two, three in text:gmatch(pattern) do
+      local line, col, warn
+      for field,position in pairs(order) do
+        local value
+        if position == 1 then
+          value = one
+        elseif position == 2 then
+          value = two
+        elseif position == 3 then
+          value = three
+        end
+
+        if field == "line" then
+          line = value
+        elseif field == "col" then
+          col = value
+        elseif field == "message" then
+          warn = value
+        end
+      end
+
+      if line == nil then line = 1 end
+      if col == nil then col = 1 end
+      if warn == nil then warn = "syntax error" end
+
+      if current_doc ~= nil then
+        if tonumber(line) > #current_doc.lines then
+          line = #current_doc.lines
+        end
+      end
+      coroutine.yield(line, col, warn)
```

Comment:
Since the values are already ordered in the ordered_fields table.
We can used the lua table.unpack to take them out of the list and use them as parameters to the yield.
```lua
table.unpack(ordered_fields)
```